packages/relay/src/lib/eth.ts

```diff
@@ -670,7 +679,8 @@ export class EthImpl implements Eth {
     let timestamp = 0;
     const transactions: Transaction[] = [];
     const transactionHashes: string[] = [];
-    contractResults.results.forEach(async (result) => {
+    for (const i in contractResults.results) {
```

Comment:
Was gonna suggest the below but rebasing should bring this in for you
```suggestion
    for (const result of contractResults.results) {
```

Reply:
ahh... `of` (values) vs `in` (indexes).  I learned new JS language today.